### PR TITLE
Ports/gltron: Use sdl12-compat

### PR DESCRIPTION
**Pull request by lightspot21**
This PR:

- Adds a port of `SDL_sound` compatible with SDL 1.x. The 2.x version of `SDL_sound` is also moved to `SDL2_sound`, for consistency with the naming of SDL2 versions of SDL helper libraries.
- Removes the patches modifying `gltron` to use SDL 2 directly.

Notes:
- The SDL_sound port supports the bare minimum for `gltron` to run: it doesn't support `libvorbis` and `libmodplug`, due to linker errors when those are enabled (specifically when `libvorbis` is used linking fails with missing libvorbis.la file although it seems to be installed under /usr/local/lib, and when using `libmodplug` ld complains about undefined functions, although libmodplug is also installed). I'd like to somehow enable these too, but I have no time to investigate the linker errors right now.

### CODE REVIEW

File: Ports/AvailablePorts.md [hunk 1]
```diff
@@ -203,10 +203,11 @@ This list is also available at [ports.serenityos.net](https://ports.serenityos.n
...
 | [`SDL2_net`](SDL2_net/)                             | SDL2\_net (network add-on for SDL2)                             | 2.0.1                    | https://www.libsdl.org/projects/SDL_net/                                       |
 | [`SDL2_ttf`](SDL2_ttf/)                             | SDL2\_ttf (TrueType Font add-on for SDL2)                       | 2.0.18                   | https://www.libsdl.org/projects/SDL_ttf/                                       |
 | [`SDL_mixer`](SDL_mixer/)                           | SDL\_mixer (Audio mixer add-on for SDL 1.2)                     | 1.2.12                   | https://www.libsdl.org/projects/SDL_mixer/release-1.2.html                     |
-| [`SDL_sound`](SDL_sound/)                           | SDL_sound (Abstract soundfile decoder)                          |                          | https://github.com/icculus/SDL_sound                                           |
+| [`SDL_sound`](SDL_sound/)                           | SDL_sound (Abstract soundfile decoder add-on for SDL 1.2)       | 1.0.3                    | https://www.icculus.org/SDL_sound/                                             |
```
Comment (timschumi) [inline Ports/AvailablePorts.md:210]: This should be re-added in the commit that actually adds the `SDL_sound` port.

File: Ports/SDL_sound/package.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='SDL_sound'
+version='1.0.3'
+useconfigure='true'
+use_fresh_config_sub='true'
+depends=("sdl12-compat" "libmikmod")
+files="https://www.icculus.org/SDL_sound/downloads/${port}-${version}.tar.gz ${port}-${version}.tar.gz 3999fd0bbb485289a52be14b2f68b571cb84e380cc43387eadf778f64c79e6df" 
+auth_type='sha256'
+configopts=("--with-sdl-prefix=${SERENITY_INSTALL_ROOT}/usr/local" "--enable-ogg=no" "--enable-modplug=no")
+makeopts=("LDFLAGS=-lm")
```
Comment (timschumi) [inline Ports/SDL_sound/package.sh:10]: As usual, please split these up across multiple lines.

File: Ports/SDL2_sound/package.sh [hunk 3]
```diff
@@ -1,12 +1,12 @@
 #!/usr/bin/env -S bash ../.port_include.sh
-port=SDL_sound
-version=git
-_commit=df3fc779774c2c5dc1147239da1af858c88f1a74
-workdir=SDL_sound-${_commit}
-useconfigure=true
-depends=("SDL2")
+port='SDL2_sound'
```
Comment (timschumi) [inline Ports/SDL2_sound/package.sh:2]: Surely, there are ports that rely on `SDL_sound`, right? Those ports dependencies should be adapted as well.

**Review by timschumi — CHANGES_REQUESTED** [review]
What was the issue with linking `libvorbis`? Is it possible that it was just missing a `--with-sysroot=${SERENITY_INSTALL_ROOT}` in its `configopts`?

File: Ports/SDL2_sound/package.sh (hunk 3 above)
Comment (lightspot21) [reply Ports/SDL2_sound/package.sh:2]:
```
[lightspot21@Workbox-MK2 Ports]$ rg SDL_sound
SDL_sound/package.sh
2:port='SDL_sound'
7:files="https://www.icculus.org/SDL_sound/downloads/${port}-${version}.tar.gz ${port}-${version}.tar.gz 3999fd0bbb485289a52be14b2f68b571cb84e380cc43387eadf778f64c79e6df"

gltron/package.sh
8:depends=("libpng" "glu" "SDL_sound" "sdl12-compat" "zlib")
13:    "LIBS=-lm -lSDL_sound"

SDL2_sound/package.sh
5:workdir="SDL_sound-${_commit}"
8:files="https://github.com/icculus/SDL_sound/archive/${_commit}.zip ${_commit}.zip bf655a03ab96a49c4140e19135433d62893c124330955e85e3dfddbe9963bac2"

SDL2_sound/patches/ReadMe.md
1:# Patches for SDL_sound on SerenityOS

AvailablePorts.md
199:| [`SDL2_sound`](SDL2_sound/)                         | SDL2\_sound (Abstract soundfile decoder add-on for SDL2)        |                          | https://github.com/icculus/SDL_sound                                           |
202:| [`SDL_sound`](SDL_sound/)                           | SDL_sound (Abstract soundfile decoder add-on for SDL 1.2)       | 1.0.3                    | https://www.icculus.org/SDL_sound/                                             |
```

Seems like nothing else apart from gltron itself needs SDL_sound, actually

Comment (timschumi) [reply Ports/SDL2_sound/package.sh:2]: Then that dependency should be renamed in this commit and only be replaced later on.

Comment (lightspot21) [reply Ports/SDL2_sound/package.sh:2]: Done

**Comment by timschumi** [comment]
> What was the issue with linking `libvorbis`? Is it possible that it was just missing a `--with-sysroot=${SERENITY_INSTALL_ROOT}` in its `configopts`?

Anything on this?

**Comment by lightspot21** [comment]
> > What was the issue with linking `libvorbis`? Is it possible that it was just missing a `--with-sysroot=${SERENITY_INSTALL_ROOT}` in its `configopts`?
> 
> Anything on this?

Oh I couldn't find time to look at it, but now I have some

You mean modify the `libvorbis` port to have `with-sysroot` set?

**Comment by timschumi** [comment]
> > > What was the issue with linking `libvorbis`? Is it possible that it was just missing a `--with-sysroot=${SERENITY_INSTALL_ROOT}` in its `configopts`?
> > 
> > 
> > Anything on this?
> 
> Oh I couldn't find time to look at it, but now I have some
> 
> You mean modify the `libvorbis` port to have `with-sysroot` set?

That completely depends on the error that you are getting. I'd lean towards "`SDL_sound` needs `--with-sysroot`" though, based on previous experience.

**Review by timschumi — APPROVED** [review]
LGTM! Issues with `libvorbis` and friends can be sorted out after i ~~finish~~ start porting `libtool` patches for sysroots to older versions.

**Review by linusg — CHANGES_REQUESTED** [review]
SDL2_sound readme still says `Patches for SDL_sound on SerenityOS`

**Comment by lightspot21** [comment]
> SDL2_sound readme still says `Patches for SDL_sound on SerenityOS`

Woops, should be fixed now

**Pull request merged**